open-sphere-base/core/src/main/java/io/opensphere/core/util/javafx/input/IntegerSpinnerInputValidator.java

```diff
@@ -0,0 +1,107 @@
+package io.opensphere.core.util.javafx.input;
+
+import org.apache.commons.lang3.StringUtils;
+
+import javafx.application.Platform;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.SpinnerValueFactory.IntegerSpinnerValueFactory;
+import javafx.scene.control.TextField;
+import javafx.util.StringConverter;
+
+/**
+ * Ensures that only integer input is entered into a integer spinner.
+ */
+public class IntegerSpinnerInputValidator extends StringConverter<Integer>
+{
+    /** The maximum allowable value in the spinner. */
+    private final int myMaximum;
+
+    /** The minimum allowable value in the spinner. */
+    private final int myMinimum;
+
+    /**
+     * Creates a new input validator for the given Spinner TextField.
+     *
+     * @param input the TextField analyzed for validation
+     * @param min the minimum value allowed
+     * @param max the maximum value allowed
+     */
+    public IntegerSpinnerInputValidator(TextField input, int min, int max)
+    {
+        myMinimum = min;
+        myMaximum = max;
+
+        // invalid input immediately resets the spinner to its last value
+        input.textProperty().addListener((obs, o, n) ->
+        {
+            if (!checkInput(n))
+            {
+                Platform.runLater(() -> input.setText("" + o));
+            }
+        });
+    }
+
+    /**
+     * Checks if the given string is allowed in the Spinner TextField.
+     * Valid input is empty strings and integers between the maximum
+     * and minimum (inclusive).
+     *
+     * @param input the text to validate
+     * @return true if the text is allowed in the TextField
+     */
+    private boolean checkInput(String input)
+    {
+        if (input != null && StringUtils.isEmpty(input))
```

Comment:
!input.equals(null)

Reply:
Since I'm only checking if the input is null, using == or != is correct here. In the case that input is null and .equals is used, an exception will be thrown.